### PR TITLE
tidb stability test main function

### PR DESCRIPTION
**Pull request by shuijing198799**
commit the main function of TiDB-operator stability test. and create 2 cluster for test.
PTLA @weekface @cwen0 @zyguan @tennix

### CODE REVIEW

File: tests/actions.go [hunk 1]
```diff
@@ -43,6 +49,11 @@ func NewOperatorActions(cli versioned.Interface, kubeCli kubernetes.Interface) O
 	}
 }
 
+const (
+	WAITMINUTE time.Duration = 10
+	WAITSECOND time.Duration = 5
```
Comment (zyguan) [inline tests/actions.go]: How about
```go
	DefaultPollInterval = 10 * time.Second
	DefaultPollTimeout  = 5 * time.Minute
```

File: tests/actions.go [hunk 2]
```diff
@@ -811,3 +851,473 @@ func checkoutTag(tagName string) error {
...
+		return fmt.Errorf("failed to launch adhoc backup job: %v, %s", err, string(res))
+	}
+	return nil
+}
+
+//restore full data backup is correct and check full data backup data is correct
+func (oa *operatorActions) CheckAdHocBackup(info *TidbClusterInfo) error {
+	glog.Infof("begin to clean adhoc backup")
+	defer func() {
+		glog.Infof("deploy clean backup end")
```
Comment (zyguan) [inline tests/actions.go]: clean backup -> check backup

File: tests/actions.go [hunk 3]
```diff
@@ -228,10 +248,26 @@ func (oa *operatorActions) CleanTidbCluster(info *TidbClusterInfo) error {
...
 
+
+	//for the test should add two clusters in a namespace, so we
+	//has to use clustername as an label.
+	sets := map[string]string {
+		label.InstanceLabelKey:		info.ClusterName,
+	}
+	var buffer bytes.Buffer
+	for k,v := range sets {
+		set := fmt.Sprintf("-l %s=%s",k , v)
```
Comment (zyguan) [inline tests/actions.go]: let sets = {"foo": "a", "bar": "b"}, you finally get a buffer with `"-l foo=a-l bar=b"`

File: tests/actions.go [hunk 4]
```diff
@@ -811,3 +851,473 @@ func checkoutTag(tagName string) error {
...
+	for k,v := range sets {
+		set := fmt.Sprintf(" --set %s=%s",k , v)
+		_,err := buffer.WriteString(set)
+		if err != nil  {
+			return err
+		}
+	}
+
+	setStr := buffer.String()
+	cmd := fmt.Sprintf("helm update %s /charts/%s/tidb-cluster %s",
```
Comment (zyguan) [inline tests/actions.go]: upgrade ?

File: tests/actions.go [hunk 5]
```diff
@@ -811,3 +851,473 @@ func checkoutTag(tagName string) error {
...
+}
+
+//restore genereted record by scheduledbackup, and make sure the data is correct
+func (oa *operatorActions) CheckScheduledBackup(info *TidbClusterInfo) error {
+	glog.Infof("begin to check scheduler backup")
+	defer func() {
+		glog.Infof("deploy check scheduler end")
+	}()
+
+	jobName := fmt.Sprintf("%s--scheduled-backup",info.ClusterName)
```
Comment (zyguan) [inline tests/actions.go]: double dash ?

File: tests/actions.go (hunk 3 above)
Comment (shuijing198799) [reply tests/actions.go]: there is only one pair in this field , I make some change .thx

File: tests/actions.go (hunk 2 above)
Comment (shuijing198799) [reply tests/actions.go]: clean is not necessary， for the CleanTidbCluster function will delete all the resources。 but check is necessary for wait the back completed

File: tests/actions.go [hunk 6]
```diff
@@ -228,10 +249,26 @@ func (oa *operatorActions) CleanTidbCluster(info *TidbClusterInfo) error {
 		}
 	}
 
+
+	//for the test should add two clusters in a namespace, so we
+	//has to use clustername as an label.
+	sets := map[string]string {
```
Comment (weekface) [inline tests/actions.go]: use `label.New().Instance().String()` instead

File: tests/actions.go [hunk 7]
```diff
@@ -228,10 +249,26 @@ func (oa *operatorActions) CleanTidbCluster(info *TidbClusterInfo) error {
...
+	for k,v := range sets {
+		set := fmt.Sprintf(" %s=%s",k , v)
+		_,err := buffer.WriteString(set)
+		if err != nil  {
+			return err
+		}
+	}
+
+	setStr := buffer.String()
 	resources := []string{"cronjobs", "jobs", "pods", "pvc"}
```
Comment (weekface) [inline tests/actions.go]: Only need to delete pvc .

File: tests/actions.go [hunk 8]
```diff
@@ -311,6 +361,22 @@ func (oa *operatorActions) CheckTidbClusterStatus(info *TidbClusterInfo) error {
 	return nil
 }
 
+func (oa *operatorActions) CreateSecret(info *TidbClusterInfo) error {
```
Comment (weekface) [inline tests/actions.go]: Remove this func from this PR. May be another pr.

File: tests/actions.go [hunk 9]
```diff
@@ -811,3 +868,470 @@ func checkoutTag(tagName string) error {
 
 	return nil
 }
+
+//scheduler backup, make sure there are serveral backup record is generated, and
```
Comment (weekface) [inline tests/actions.go]: Please remove all the backup related codes from this PR. It should be another PR.

File: .gitignore [hunk 10]
```diff
@@ -7,6 +7,9 @@ images/tidb-operator-e2e/tidb-operator/
 tmp/
 data/
 .idea
+cscope.files
```
[the remark this answers is not in the document]
Comment (shuijing198799) [reply .gitignore]: it's ctags generate files only use for me

File: pkg/label/label.go [hunk 11]
```diff
@@ -138,3 +140,25 @@ func (l Label) LabelSelector() *metav1.LabelSelector {
 func (l Label) Labels() map[string]string {
 	return l
 }
+
+//Labels convers label to string
```
Comment (weekface) [inline pkg/label/label.go]: // String converts label to a string

File: pkg/label/label.go [hunk 12]
```diff
@@ -138,3 +140,25 @@ func (l Label) LabelSelector() *metav1.LabelSelector {
...
+func (l Label) String() (string,error) {
```
Comment (weekface) [inline pkg/label/label.go]: gofmt

Comment (weekface) [reply pkg/label/label.go]:
```
func (l Label) String() string {
	var arr []string

	for k, v := range l {
		arr = append(arr, fmt.Sprintf("%s=%s", k, v))
	}

	return strings.Join(arr, ",")
}

```

File: tests/actions.go [hunk 13]
```diff
@@ -228,10 +246,17 @@ func (oa *operatorActions) CleanTidbCluster(info *TidbClusterInfo) error {
 		}
 	}
 
-	resources := []string{"cronjobs", "jobs", "pods", "pvc"}
+	//for the test should add two clusters in a namespace, so we
```
Comment (weekface) [inline tests/actions.go]: delete these lines.

File: pkg/label/label.go [hunk 14]
```diff
@@ -14,8 +14,10 @@
 package label
 
 import (
+	"fmt"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"strings"
```
Comment (cwen0) [inline pkg/label/label.go]:
```
import (
       "fmt"
        "strings" 

         metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
         "k8s.io/apimachinery/pkg/labels"
)
``` 
format code

File: tests/actions.go [hunk 15]
```diff
@@ -204,6 +212,10 @@ func (oa *operatorActions) DumpAllLogs(info *OperatorInfo, clusterInfo *TidbClus
 }
 
 func (oa *operatorActions) DeployTidbCluster(info *TidbClusterInfo) error {
+	glog.Infof("begin to deploy tidb cluster")
```
Comment (xiaojingchen) [inline tests/actions.go]: print more info about the cluster, like `name` and `namespace`

File: tests/actions.go [hunk 16]
```diff
@@ -215,10 +227,13 @@ func (oa *operatorActions) DeployTidbCluster(info *TidbClusterInfo) error {
 }
 
 func (oa *operatorActions) CleanTidbCluster(info *TidbClusterInfo) error {
+	glog.Infof("begin to clean tidb cluster")
```
Comment (xiaojingchen) [inline tests/actions.go]: ditto
